Engine/DataFeeds/SubscriptionFrontierTimeProvider.cs

```diff
@@ -57,12 +57,20 @@ private void UpdateCurrentTime()
             long earlyBirdTicks = MaxDateTimeTicks;
             foreach (var subscription in _subscriptionManager.DataFeedSubscriptions)
             {
-                if (subscription.Current == null
+                // this if should just be 'subscription.Current == null' but its affected by GH issue 3914
+                if (// this is a data subscription we just added
+                    // lets move it next to find the initial emit time
+                    subscription.Current == null
```

Comment:
per the `IEnumerator<T>` spec, `IEnumerator<T>.Current` is not defined _before_ a call to `MoveNext()` has happened and the framework implementations will throw an `InvalidOperationException`. I think our convention has been to initialize `Current` to null for data enumerators, so I think this is safe for our application, but wanted to make note of the broken contract here. This (and a few other instances) make me think LEAN would benefit from defining it's own `IEnumerator<BaseData>` type. The goal would be to support the same style of composition (decorators/russian dolls), but singling out data enumerators as a first-class citizen in LEAN's object model.